### ISSUE TITLE
[bitnami/postgresql-ha] select pg_reload_conf() don't work 

### ISSUE THREAD

**Issue by Mihai-CMM** [bitnami/charts#13676]
### Name and Version

bitnami/postgresql-ha

### What steps will reproduce the bug?

override CM is only applied at pod creation
```
`  override.conf: |-
    shared_buffers = 4194304
    work_mem = 655MB
    maintenance_work_mem = 8GB
    effective_io_concurrency = 200
    wal_buffers = 64MB
    min_wal_size = 1GB
    effective_cache_size = 96GB
    autovacuum_max_workers = 8
    max_worker_processes = 16
    max_parallel_workers = 16
    max_parallel_workers_per_gather = 9 ## note this new value ###
    max_parallel_maintenance_workers = 8
    max_wal_size = 12GB
`
```
```
postgres=# SELECT  name,setting FROM pg_settings WHERE name LIKE '%work%';
               name                | setting
-----------------------------------+---------
 autovacuum_max_workers            | 8
 autovacuum_work_mem               | -1
 logical_decoding_work_mem         | 65536
 maintenance_work_mem              | 8388608
 max_logical_replication_workers   | 4
 max_parallel_maintenance_workers  | 8
 max_parallel_workers              | 16
 max_parallel_workers_per_gather   | 8 ## this was initial cm set value ###
 max_sync_workers_per_subscription | 2
 max_worker_processes              | 16
 recursive_worktable_factor        | 10
 work_mem                          | 670720
(12 rows)
```

after running 

```
postgres=# select pg_reload_conf();
 pg_reload_conf
----------------
 t
(1 row)
```

the value from CM is not updated 

```
SELECT  name,setting FROM pg_settings WHERE name LIKE '%work%';
               name                | setting
-----------------------------------+---------
 autovacuum_max_workers            | 8
 autovacuum_work_mem               | -1
 logical_decoding_work_mem         | 65536
 maintenance_work_mem              | 8388608
 max_logical_replication_workers   | 4

 max_parallel_maintenance_workers  | 8
 max_parallel_workers              | 16
 max_parallel_workers_per_gather   | 8  ### still old value ###
 max_sync_workers_per_subscription | 2
 max_worker_processes              | 16
 recursive_worktable_factor        | 10
 work_mem                          | 670720

```

### Are you using any custom parameters or values?

in values.yaml I have this entry that generates the CM

```
  extendedConf: |-
    shared_buffers = 4194304
    work_mem = 655MB
    maintenance_work_mem = 8GB
    effective_io_concurrency = 200
    wal_buffers = 64MB
    min_wal_size = 1GB
    effective_cache_size = 96GB
    autovacuum_max_workers = 8
    max_worker_processes = 16
    max_parallel_workers = 16
   max_parallel_workers_per_gather  = 8
    max_parallel_maintenance_workers = 8
    max_wal_size = 12GB
   
```

### What is the expected behavior?

to be able to operate the psql cluster and set settings using the config map and pg_reload

### What do you see instead?

per above the config in cm is not applied by pg_reload

**Comment by rafariossaa** [comment]
> override CM is only applied at pod creation
Hi,
 Are you deploying the chart with the configuration or are you changing the configuration and restarting the pods ?

  I am asking, because configuration is only applied the when deploying the chart for first time. The initialization scripts in the image container will skip any configuration related procedures if persisted data is detected.

**Comment by Mihai-CMM** [comment]
Hello

I am applying the chart with the configuration to be in proper "gitops" sync. 
The problem is that in fact I cannot operate the cluster if the configmap changes cannot be applied without full helm redeployment.

My goal is to have the initial tune values in the chart but after I gather 10 TB of data to be able to simply do a pg_reload when i change the number of works like in the example above.
The issue with re running chart with huge amount of data was subject to a different issue (#13677) created and pg_rewind helped but I am not sure on the long term that I will be able to use it.

In short is there a way to reload postgres config from the configmaps new entries?
On the other hand if I run a psql cli like "set max number of workers" from inside postgres ,that sticks though is not persistent so I am going back to configmap ?

Hope I am clear on how I "think" it should work?

Thanks

**Comment by rafariossaa** [comment]
Hi,
  Thanks for describing your use case. 
  
> In short is there a way to reload postgres config from the configmaps new entries?
I think you could use: `pg_ctl reload`

> On the other hand if I run a psql cli like "set max number of workers" from inside postgres ,that sticks though is not persistent so I am going back to configmap ?
Have you tried upgrading the chart ?
I tested it deploying the default chart with no changes in the values.yaml file:
```bash
$ helm install mypg -f values.yaml .
```
and once deployed, I edited values.yaml and added:
```yaml
  extendedConf: |-
     max_parallel_maintenance_workers=10
```
then, I upgraded the chart using:
```bash
$ helm upgrade mypg -f values.yaml --set pgpool.adminPassword=$ADMIN_PASSWORD --set postgresql.repmgrPassword=$REPMGR_PASSWORD --set postgresql.password=$PASSWORD .
```
Now, I got the parameter updated (the default value was: 2):
```bash
$ kubectl run mypg-postgresql-ha-client --rm --tty -i --restart='Never' --namespace default --image docker.io/bitnami/postgresql-repmgr:15.1.0-debian-11-r0 --env="PGPASSWORD=$POSTGRES_PASSWORD"          --command -- psql -h mypg-postgresql-ha-pgpool -p 5432 -U postgres -d postgres
If you don't see a command prompt, try pressing enter.

postgres=# show max_parallel_maintenance_workers ;
 max_parallel_maintenance_workers 
----------------------------------
 10
(1 row)
```

  In any case, I am creating an internal task to review this kind of use case and improve the user experience.
  We are a small team and have to deal with deadlines and priorities, hence I can not provide an ETA, but we will come back as soon as we have news.
